doc/controller-card_poweroff.md

```diff
@@ -0,0 +1,132 @@
+
+    **Response:**
+
+
+	Response here is the same as "1" above.
+
+
+
+3. Both controller-cards are functional and the the secondary controller card is attempted to be powered off.
+
+    **Response:**
+
+
+    The shutdown can be initiated by pushing the command [/components/component[controller-card#]/controller-card/config/power-admin-state = POWER\_DISABLED](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-platform.html#components-component-controller-card-config-power-admin-state) to the box. As a result, the secondary controller card whose [/components/component/state/redundant-role](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-platform.html#components-component-state-redundant-role) is SECONDARY
+
+
+    will be shutdown and its [/components/component[controller-card#]/controller-card/state/power-admin-state](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-platform.html#components-component-controller-card-state-power-admin-state) will change to "POWER\_DISABLED". This configuration is expected to stay sticky in the box and when the device reboots, the subject controller-card is expected to remain Powered-off.
+
+4. Both controller-cards are functional and the the primary controller card is attempted to be shutdown
+
+    **Response:**
+
+
+    Follow Rule#1 above
+
+5. Only one controller-card is present and the card is attempted to be shutdown.
+
+    **Response:**
+
+
+    Since the lone controller-card is PRIMARY, please follow Rule#1 above.
+
+6. Faulty card is shutdown and then replaced with a working card but device still has the config of `/components/component[controller-card#]/controller-card/config/power-admin-state = POWER_DISABLED`
+
+    **Response:**
+
+
+    This is as per expectations so inserting a new card into the chassis can happen anytime. However, bringing it online happens only in a controlled environment using a configuration change to set `/components/component[controller-card#]/controller-card/config/power-admin-state = POWER_ENABLED`
+
+
+7. Both controller-cards present during the initial boot operation. However, the controller card that is configured as "POWER\_DISABLED" boots up sooner than the other card and takes over the ACTIVE role.
+
+    **Response:**
+
+
+    The controller cards should never take PRIMARY/SECONDARY roles unless the configuration is fully loaded and processed. Hence, once the configuration is loaded, the card configured to be "POWER\_DISABLED" should shutdown. The only exception is when it is the only controller card in the chassis. In which case, the controller card can still load irrespective of its power-admin-state being POWER\_DISABLED. In this situation though, the implementation must send a Syslog message of the severity "Warning" to inform the Operator about the situation.
```

Comment:
This exception doesn't apply to this scenario because we've said `Both controller-cards present during the initial boot operation`

However, we could add another scenario below where:

Two cards are present, secondary is powered off.  The primary card is removed and the system goes offline.  If the device is rebooted (for example via externally removing and reapplying power ), what should happen?

Reply:
`This exception doesn't apply to this scenario because we've said Both controller-cards present during the initial boot operation`

Well, not necessarily for all vendors today config load happens before the PRIMARY/SECONDARY election process. Sequence is indeterministic today between config load operation and election process and hence the scenario here is very well a possibility and is orthogonal to the presence of both cards or not. We are calling out this scenario to ensure expectations are clear.

We can definitely add the scenario you mentioned.

Reply:
The scenario is defined as two controller cards are present, but the the exception notes is:  `only exception is when it is the only controller card in the chassis. `.  Just a logic issue / should be broken out to a separate scenario IMO.

Reply:
I added Rule#5. Ptal

Reply:
LGTM.  The rule reference is a little ambiguous.  There are a couple of things in the document labeled `5.` and it's not clear which one is a rule versus a scenario.  One way to fix this would be to use outline numbered headings like:

1. Item
1.1 Item subheading
2. Another item
2.2 Another subheading